src/Raven.Server/Utils/ReplicationUtils.cs

```diff
@@ -13,6 +15,35 @@ namespace Raven.Server.Utils
 {
     internal static class ReplicationUtils
     {
+        private static readonly ConcurrentDictionary<string, (X509Certificate2, ClusterRequestExecutor)> _requestExecutorsCache = new ConcurrentDictionary<string, (X509Certificate2, ClusterRequestExecutor)>();
```

Comment:
```suggestion
        private static readonly ConcurrentDictionary<(string Url, X509Certificate2 Cert),  Lazy<ClusterRequestExecutor>> _requestExecutorsCache = new ConcurrentDictionary<(string Url, X509Certificate2 Cert),  Lazy<ClusterRequestExecutor>>();
```

Reply:
This needs calculating hash of string and object. Why not use 'string url' so we use string comperator and we have single certificate per url any how, so this way it is more efficient lookup. No?